distribution/archives/build.gradle

```diff
@@ -59,8 +59,14 @@ CopySpec archiveFiles(CopySpec modulesFiles, String distributionType, String pla
         with binFiles(distributionType, oss, jdk)
       }
       if (jdk) {
-        into('jdk') {
-          with jdkFiles(project, platform)
+        if ("darwin".equals(platform)) {
```

Comment:
This could just be `into("darwin".equals(platform) ? 'jdk.app' : 'jdk')`.

Reply:
I pushed 5b2d681ac5178f7cd93830d9c80112c0087471da directly to master and 90d44375136e0b66608be59949ab2bad4710111a to 7.x.